Refactoring to support testing TableView

kant2002 opened the pull request:
Most changes whcih propagate across codebase is related to importing JS Signals as a module which lead to conversion of modules to CommonJS. One of the module which become converted to CommonJS was cardsHelper which require remove cards globals for closed cards and making them table place specific variables which setup on game type change.

kant2002 committed:
Refactoring to support testing TableView
Most changes whcih propagate across codebase is related to importing JS Signals as a module which lead to conversion of modules to CommonJS. One of the module which become converted to CommonJS was cardsHelper which require remove cards globals for closed cards and making them table place specific variables which setup on game type change.

mtutynina commented:
Текущие ошибки в консоле игрока:
1. 
```
     Uncaught ReferenceError: signals is not defined
    at new App (application.js:14936)
    at bootstrap (application.js:16936)
    at Object.<anonymous> (application.js:17000)
    at Object.execCb (require.js:1658)
    at Module.check (require.js:874)
    at Module.<anonymous> (require.js:1121)
    at require.js:132
    at require.js:1164
    at each (require.js:57)
    at Module.emit (require.js:1163)
```
2. ```Bind popup tableMenu failed```

3. 
```
      Detailed exception:  ReferenceError: Unable to process binding "if: function (){return allowTickets }"
       Message: Unable to process binding "ltext: function (){return standupText }"
       Message: app is not defined
    at application.js:4854
    at Function.Uc (knockout-3.4.2.js:52)
    at Function.Vc (knockout-3.4.2.js:51)
    at Function.U (knockout-3.4.2.js:51)
    at Function.ua (knockout-3.4.2.js:53)
    at Function.Y (knockout-3.4.2.js:36)
    at Function.yc (knockout-3.4.2.js:50)
    at Object.W (knockout-3.4.2.js:8)
    at Object.sc (knockout-3.4.2.js:40)
    at e (knockout-3.4.2.js:48)```

kant2002 committed:
Add missing import of module "signals"

kant2002 committed:
Add missing iniitalization of back cards

kant2002 committed:
Fix test failure on Linux

kant2002 committed:
Make failing test for pot limit

kant2002 committed:
Add no limit game test

kant2002 committed:
Make first passable test for no-limit games

kant2002 committed:
Migrate gameplay tests from Chutzpah to Jest

kant2002 committed:
Add tests for the components

kant2002 committed:
Implement test for the pot limit rules

kant2002 committed:
Reduce requirements for the coverage, so PR could pass
Since we have larger set of files to cover, we will have to increase code coverage gradually until reach resonable values of 90-95%

mtutynina commented on js/table/gameactionsqueue.ts:
```diff
@@ -108,7 +118,13 @@ export class GameActionsQueue {
+            return Promise.resolve().then(() => {
+                try {
+                    callback();
+                } catch {
```
Ошибка в конструкции try catch -> catch(e) из за этого не работает tsc. И почему билд на travis ci проходит? Ты убрал tsc из билда?

kant2002 replied:
1. Какая у тебя установлена версия tsc?
2. Травис запускает через `npm run compile`

mtutynina replied:
tsc Version 2.4.2

kant2002 replied:
`npm run compile` локально работает?

mtutynina replied:
да

mtutynina commented on js/table/tableview.ts:
```diff
@@ -272,13 +275,20 @@ export class TableView {
         }, this);
+        this.gameType.subscribe((gameType) => {
+            this.places().forEach((place) => {
+                place.BackCards(gameType === 1 ? allBacksClassesTwoCards : allBacksClassesFourCards);
```
При запуске проекта не приходят BackClasses карты на стол, вместо карт просто пусто не важно какой тип игры, при этом после того как один раз сменишь тип игры карты уже приходят или если обновишь страницу.

mtutynina replied:
При этом в консоле игрока после запуска и при смене типа игры не меняются стили, если перезагрузить страницу то в дальнейшем любая смена типа игры отображает верные стили

mtutynina reviewed (changes requested):
За столом два игрока, один в статусе отошел - меняем тип игры на омаху, обновляем страницу, получаем ошибку 
```
TypeError: Cannot read property 'reduce' of null
    at ActionBlock.getPot (application.js:8076)
    at TableView.<anonymous> (application.js:8951)
    at Function.Uc (knockout-3.4.2.js:52)
    at Function.Vc (knockout-3.4.2.js:51)
    at Function.U (knockout-3.4.2.js:51)
    at Function.ec (knockout-3.4.2.js:50)
    at Function.notifySubscribers (knockout-3.4.2.js:37)
    at Function.ha (knockout-3.4.2.js:41)
    at TableView.c [as pots] (knockout-3.4.2.js:40)
    at TableView.onTableStatusInfo (application.js:9578)
```

kant2002 committed:
Update project to compile using TSC 2.6

kant2002 committed:
Fix runtime error when game is not started on the table

kant2002 committed:
Add missing Jest setup files

kant2002 committed:
Fix missed popolation of back cards on table initialization and sitting

kant2002 committed:
Increase required lines coverage
